modules/common/systemd/base.nix

```diff
@@ -0,0 +1,295 @@
+# Copyright 2022-2024 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}: let
+  # Ghaf systemd config
+  cfg = config.ghaf.systemd;
+
+  # Override minimal systemd package configuration
+  package =
+    pkgs.systemdMinimal.override {
+      pname = cfg.withName;
+      withAcl = true;
+      withAnalyze = cfg.withDebug;
+      inherit (cfg) withApparmor;
+      inherit (cfg) withAudit;
+      withCompression = true;
+      withCoredump = cfg.withDebug || cfg.withMachines;
+      inherit (cfg) withCryptsetup;
+      inherit (cfg) withEfi;
+      withBootloader = cfg.withEfi; # systemd-boot fails if not explicity set
+      inherit (cfg) withFido2;
+      inherit (cfg) withHostnamed;
+      withImportd = cfg.withMachines;
+      withKexectools = false;
+      withKmod = true;
+      withLibBPF = true;
+      withLibseccomp = true;
+      inherit (cfg) withLocaled;
+      inherit (cfg) withLogind;
+      withMachined = cfg.withMachines;
+      inherit (cfg) withNetworkd;
+      inherit (cfg) withNss;
+      withOomd = true;
+      withPam = true;
+      inherit (cfg) withPolkit;
+      inherit (cfg) withResolved;
+      withShellCompletions = cfg.withDebug;
+      withTimedated = true;
+      inherit (cfg) withTimesyncd;
+      inherit (cfg) withTpm2Tss;
+      withUtmp = cfg.withJournal || cfg.withAudit;
+    } # To be removed, current systemd version 254.6 < 255
+    // lib.optionalAttrs (lib.hasAttr "withVmspawn" (lib.functionArgs pkgs.systemd.override)) {
+      withVmspawn = cfg.withMachines;
+    };
+
+  # Definition of suppressed system units in systemd configuration. This removes the units and has priority.
+  # Required to avoid build failures compared to only disabling units for some options. Note that errors will be silently ignored.
```

Comment:
Could there be a way to log the errors with `-debug` builds?

Reply:
No I don't think so - the units are filtered with the list of suppressedUnits, so any string therein will work without errors.
`enabledUnits = filterAttrs (n: v: ! elem n cfg.suppressedSystemUnits) cfg.units;`

Reply:
No I don't think so - the units are filtered with the list of suppressedUnits, so any string therein will work without errors.
`enabledUnits = filterAttrs (n: v: ! elem n cfg.suppressedSystemUnits) cfg.units;`